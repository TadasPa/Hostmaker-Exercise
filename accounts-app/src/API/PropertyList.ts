import APIUrls from "./Urls"

class PropertyListAPI {

    static retrievePropertyList() {
        const url = APIUrls.PropertyListUrl;
    
        return fetch(url, {
          method: "GET",
          headers: {
            //for auth or similar
          }
        }).then(httpResponse => {
          if (httpResponse.status !== 200) {
            return null;
          }
          const propertyList = httpResponse.json();
          return propertyList;
        },
          error => {
            return null;
          });
      }    

}

export default PropertyListAPI;