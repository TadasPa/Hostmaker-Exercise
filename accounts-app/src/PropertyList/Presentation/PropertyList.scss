.property-list-presentation-container {
    padding: 1rem;
    color: rgba(73,73,80,.8);
}

.property-item-row {
    display: flex;
    border-top-style: solid;
    border-color: #2ca9a5;
    border-width: 0.01rem;
    padding-top: 2rem;
    padding-bottom: 2rem;
}

.property-item-row-title {
    display: flex;    
    padding-top: 2rem;
    padding-bottom: 2rem;
    //font-weight: 700;
    color: #494950;
    opacity: .7;
}

.property-item-owner {
    width: 30rem;
    justify-content: flex-end;
    display: flex;
    align-items: flex-start;
}

.property-item-address {
    width: 30rem;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    padding-left: 10rem;
}

.property-item-income-generated {
    width: 30rem;
    justify-content: flex-start;
    display: flex;
    align-items: flex-start;
}

.section-title {
    color: #333;
    padding: 1rem;
    font-size: 2rem;
}

.property-table {
    padding-left: 6rem;
    padding-right: 6rem;
}