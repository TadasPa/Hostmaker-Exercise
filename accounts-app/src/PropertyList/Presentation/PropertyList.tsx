import React, { Component } from "react";
import './PropertyList.css';
import { PropertyListState } from "../State/Reducer";

export interface IPropertyListPresentationProps extends PropertyListState {
    fetchData: () => void
}

export class PropertyListPresentation extends Component<IPropertyListPresentationProps> {

    componentDidMount() {
        this.props.fetchData();
    }

    render() {
        const propertyList = this.props.propertyList.map(property => {
            return (
                <div key={property.airbnbId}>
                    Owner: {property.owner}
                </div>
            );
        });

        return (
            <div className="property-list-presentation-container">
                {this.props.isLoading && "Loading..."}
                {this.props.loadError && this.props.loadError}
                {propertyList}
            </div>
        );
    }
}