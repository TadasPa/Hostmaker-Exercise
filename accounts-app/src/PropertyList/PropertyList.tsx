import React, { Component } from "react";
import './PropertyList.scss';
import { connect } from "react-redux";
import { PropertyListState } from "./State/Reducer"
import { PropertyListActions } from "./State/Actions";
import { PropertyListPresentation, IPropertyListPresentationProps } from "./Presentation/PropertyList";

export class PropertyListContainer extends Component<IPropertyListPresentationProps> {

    render() {
        return (
            <div className="property-list-container">
                <PropertyListPresentation
                    isLoading={this.props.isLoading}
                    propertyList={this.props.propertyList}
                    loadError={this.props.loadError}
                    fetchData={this.props.fetchData} />
            </div>
        );
    }
}

const mapStateToProps = (state: { PropertyList: PropertyListState }) => {
    
    return {
        isLoading: state.PropertyList.isLoading,
        propertyList: state.PropertyList.propertyList,
        loadError: state.PropertyList.loadError
    }
}

const mapDispatchToProps = (dispatch: any) => {
    return {
        fetchData: () => {
            dispatch(PropertyListActions.fetchData());
        }
    }
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(PropertyListContainer)