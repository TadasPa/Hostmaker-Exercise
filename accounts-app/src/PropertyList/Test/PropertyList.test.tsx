import React from 'react';
import { shallow } from 'enzyme';
import {PropertyListContainer} from '../PropertyList';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
     shallow(<PropertyListContainer isLoading={false} propertyList={[]} loadError={""} fetchData={()=>{}} />);
});

it('render defaults', () => {
     const propertyList = [
          {
               "owner": "33 miles away",
               "address": {
                   "line1": "",
                   "line4": "",
                   "postCode": "SG4 7JD",
                   "city": "London",
                   "country": "U.K."
               },
               "airbnbId": 3512503,
               "numberOfBedrooms": 1,
               "numberOfBathrooms": 1,
               "incomeGenerated": 2000.34
             }
     ];
     const propertyListContainer = renderer
          .create(<PropertyListContainer isLoading={false} propertyList={propertyList} loadError={""} fetchData={jest.fn()} />)
          .toJSON();
     expect(propertyListContainer).toMatchSnapshot();
});