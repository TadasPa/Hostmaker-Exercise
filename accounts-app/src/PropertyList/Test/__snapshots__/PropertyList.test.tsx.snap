// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render defaults 1`] = `
<div
  className="property-list-container"
>
  <div
    className="property-list-presentation-container"
  >
    <div>
      <img
        alt="Hostmaker logo"
        src="https://hostmaker-website.imgix.net/assets/images/logos/hostmaker-logo.svg?auto=compress%2Cformat&fit=crop&ixlib=react-8.4.0&h=38&w=200"
      />
    </div>
    <div
      className="property-table"
    >
      <div
        className="property-item-row-title"
      >
        <div
          className="property-item-owner"
        >
          Owner
        </div>
        <div
          className="property-item-address"
        >
          Address
        </div>
        <div
          className="property-item-income-generated"
        >
          Income generated
        </div>
      </div>
      <div
        className="property-item-row"
      >
        <div
          className="property-item-owner"
        >
          33 miles away
        </div>
        <div
          className="property-item-address"
        >
          <div>
            
          </div>
          <div />
          <div />
          <div>
            
          </div>
          <div>
            SG4 7JD
          </div>
          <div>
            London
          </div>
          <div>
            U.K.
          </div>
        </div>
        <div
          className="property-item-income-generated"
        >
          2000.34
        </div>
        <div
          className="service-area-marker-container"
        >
          <div
            className="service-area-marker-unknow-state"
            title="Service area is beeing checked at the moment."
          >
            <svg>
              <path
                clipRule="evenodd"
                d="M7 .01C3.144.01.008 3.148.008 7.004c0 4.483 5.967 11.765 6.22 12.072l.772.936.771-.936c.254-.308 6.221-7.59 6.221-12.072C13.992 3.147 10.856.01 7 .01zm0 2a5 5 0 0 1 4.992 4.994c0 2.705-3.114 7.357-4.992 9.822-1.878-2.463-4.992-7.113-4.992-9.822A5 5 0 0 1 7 2.01zM4.5 7a2.5 2.5 0 1 1 5 0 2.5 2.5 0 0 1-5 0z"
                fill="#1F242B"
                fillRule="evenodd"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    
  </div>
</div>
`;
